### PR TITLE
Adds drawing chalk outlines on dead mobs, white crayons are now called sticks of chalk

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request
image
*The Miami Mutilator is at large once again. Thank god we have such hardboiled detectives as Manny Pardo on the case.*

This pull request allows for players to use white crayons, which have now been renamed to sticks of chalk, to generate outlines of dead or "dead" bodies found throughout the station in an attempt to let detectives actually set up crime scenes as opposed to, more typically, just causing them.

Players will attempt to draw a chalk outline if the target is a mob and the mob is dead (or fake-dead), using a single charge of the crayon/chalk. Also, adds a quick macro so that we can get the proper left/right orientation of the dead mob for reference.

## Why It's Good For The Game

Detective should have more thematic, simple ways to organize crime scenes in-round, and what better way to do that then through existing items and mechanics that we already have. Also, I was 100% shocked to learn that these have always been white crayons as opposed to sticks of chalk, which certainly seems odd. In essence, this change doesn't really let you do anything "new", but just makes it easier and simpler to do so in a regular round without nearly as much fiddling.

This mechanic exists exclusively within the afterattack of sticks of chalk as opposed to all crayons for thematic reasons, but I'm not 100% sold on that and it would be simple to move over to all crayons if people feel strongly about that.

## Changelog
:cl:
add: White crayons (Renamed to Sticks of Chalk) may now be used on dead bodies to draw a body outline onto the ground easily.
/:cl:

### CODE REVIEW

File: code/game/objects/items/crayons.dm [hunk 1]
```diff
@@ -645,13 +645,37 @@
...
+			return NONE
+
+		var/decal_rotation = GET_LYING_ANGLE(pwned_human) - 90
+		var/obj/effect/decal/cleanable/crayon/chalk_line = new(get_turf(pwned_human), paint_color, "body", "chalk outline", decal_rotation, null, "A vaguely [pwned_human] shaped outline of a body.")
+		to_chat(user, span_notice("You draw a chalk outline around [pwned_human]."))
+		chalk_line.pixel_y = (pwned_human.pixel_y + pwned_human.pixel_z) + rand(-2, 2)
+		chalk_line.pixel_x = (pwned_human.pixel_x + pwned_human.pixel_w) + rand(-1, 1)
+		return ITEM_INTERACT_SUCCESS
+	return ..()
+
```
Comment (ZephyrTFA) [inline code/game/objects/items/crayons.dm:678]:
```suggestion
/obj/item/toy/crayon/white/interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
	/// Wherein, we draw a chalk body outline vaguely around the dead or "dead" mob
	if(!ishuman(interacting_with) || user.combat_mode)
		return ..()

	var/mob/living/carbon/human/pwned_human = interacting_with

	if(!(pwned_human.stat == DEAD || HAS_TRAIT(pwned_human, TRAIT_FAKEDEATH)))
		balloon_alert_to_viewers("FEEDING TIME")
		return ..()

	balloon_alert_to_viewers("drawing outline...")
	if(!do_after(user, DRAW_TIME, target = pwned_human, max_interact_count = 4))
		return NONE
	if(!use_charges(user, 1))
		return NONE

	var/decal_rotation = GET_LYING_ANGLE(pwned_human) - 90
	var/obj/effect/decal/cleanable/crayon/chalk_line = new(get_turf(pwned_human), paint_color, "body", "chalk outline", decal_rotation, null, "A vaguely [pwned_human] shaped outline of a body.")
	to_chat(user, span_notice("You draw a chalk outline around [pwned_human]."))
	chalk_line.pixel_y = (pwned_human.pixel_y + pwned_human.pixel_z) + rand(-2, 2)
	chalk_line.pixel_x = (pwned_human.pixel_x + pwned_human.pixel_w) + rand(-1, 1)
	return ITEM_INTERACT_SUCCESS

```

**Comment by SomeoneYouProbablyKnow** [comment]
Hell yea, sadly this won't happen as if there's a body you're gonna revive and simply ask them but hell yea anyway

**Comment by AdipemDragon** [comment]
this is me not having played in a bit, but won't this affect mimes? since, iirc, their pda pen Is A White Crayon

**Comment by ArcaneMusic** [comment]
> this is me not having played in a bit, but won't this affect mimes? since, iirc, their pda pen Is A White Crayon

As a matter of fact, the mime black/white crayon is a different subtype than the white crayon, but I did confirm that ahead of time:
`/obj/item/toy/crayon/mime` vs `/obj/item/toy/crayon/white` vs `/obj/item/toy/crayon/black`

**Pull request merged**